src/components/swap/TradePrice.tsx

```diff
@@ -33,7 +33,14 @@ export default function TradePrice({ price }: TradePriceProps) {
   const { baseCurrency, quoteCurrency } = price
   const { data: usdPrice } = useUSDPrice(tryParseCurrencyAmount('1', showInverted ? baseCurrency : quoteCurrency))
 
-  const formattedPrice = formatPrice(showInverted ? price : price.invert(), NumberType.TokenTx)
+  let formattedPrice: string
+  try {
+    formattedPrice = showInverted
+      ? formatPrice(price, NumberType.TokenTx)
+      : formatPrice(price.invert(), NumberType.TokenTx)
```

Comment:
nit: the old syntax would cut down on duplicated code
```suggestion
    formattedPrice = formatPrice(showInverted ? price : price.invert(), NumberType.TokenTx)
```